File: Task2/Models/Series.swift
//
//  Series.swift
//  Task2
//
//  Created by skia mac mini on 5/8/24.
//

import Foundation

struct Series: Decodable {
    
    let id: Int
    let dicomStudyId: Int
    let seriesInstanceUID: String
    let seriesNumber: String
    let modality: String
    let seriesDateTime: String
    let patientPosition: String
    let acquisitionNumber: String
    let scanningSequence: String?
    let bodyPartExamined: String
    let seriesDescription: String
    let rows: String
    let columns: String
    let pixelSpacing: String
    let sliceThickness: String
    let imageType: String
    let imageOrientationPatient: String
    let numberOfDicomFiles: Int
    let createdDateTime: String
    let lastModifiedDateTime: String
    let volumeFilePath: String
    let tag: String?
    let memo: String?
    
}

File: server/DicomServer/Models/Series.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DicomServer.Models
{
    struct Series 
    {
        int id { get; }
        int dicomStudyId { get; }
        string seriesInstanceUID { get; }
        string seriesNumber { get; }
        string modality { get; }
        string seriesDateTime { get; }
        string patientPosition { get; }
        string acquisitionNumber { get; }
        string ? scanningSequence { get; }
        string bodyPartExamined { get; }
        string seriesDescription { get; }
        string rows { get; }
        string columns { get; }
        string pixelSpacing { get; }
        string sliceThickness { get; }
        string imageType { get; }
        string imageOrientationPatient { get; }
        int numberOfDicomFiles { get; }
        string createdDateTime { get; }
        string lastModifiedDateTime { get; }
        string volumeFilePath { get; }
        string ? tag { get; }
        string ? memo { get; }
    }
}